base/array.jl

```diff
@@ -320,6 +320,8 @@ oneunit(x::AbstractMatrix{T}) where {T} = _one(oneunit(T), x)
 
 convert(::Type{Vector}, x::AbstractVector{T}) where {T} = convert(Vector{T}, x)
 convert(::Type{Matrix}, x::AbstractMatrix{T}) where {T} = convert(Matrix{T}, x)
+convert(::Type{Matrix}, x::Vector{T}) where {T} = Matrix(reshape(x, :, 1))
+convert(::Type{Matrix}, x::RowVector{T,S}) where {T,S} = Matrix(reshape(x, 1, :))
```

Comment:
He means combinations of `reshape` and `resize!` behave funny (sorry there's a bit of OhMyREPL going on below):
```
julia 0.5> x = Vector(1:4)
4-element Array{Int64,1}:
 1
 2
 3
 4

julia 0.5> y = reshape(x, 2, 2)
2×2 Array{Int64,2}:
 1  3
 2  4

julia 0.5> resize!(x, 5)
------ ErrorException ------------------ Stacktrace (most recent call last)

 [1] — resize!(::Array{Int64,1}, ::Int64) at array.jl:512

cannot resize array with shared data
```
The implementation of `Array` is a bit "special". While it's always been semantically allowed to resize a vector, a reshaped array will by default be a view of the original data. So if you have two views of the same data, and resize one, you want to guard against the fact the second data semantically hasn't been resized. Either we make this an error or else *sometimes* transparently make a copy in circumstances that the author of a local method can't predict.